src/LinuxTracing/LinuxTracingIntegrationTest.cpp

```diff
@@ -0,0 +1,300 @@
+  }
+
+  [[nodiscard]] std::string ReadLine() {
+    std::string str;
+    while (true) {
+      char c;
+      CHECK(read(reading_fd_, &c, sizeof(c)) == sizeof(c));
+      if (c == '\n' || c == '\0') break;
+      str.push_back(c);
+    }
+    return str;
+  }
+
+ private:
+  pid_t child_pid_ = -1;
+  int reading_fd_ = -1;
+  int writing_fd_ = -1;
+};
+
+class BufferTracerListener : public TracerListener {
+ public:
+  void OnSchedulingSlice(orbit_grpc_protos::SchedulingSlice scheduling_slice) override {
+    orbit_grpc_protos::CaptureEvent event;
+    *event.mutable_scheduling_slice() = std::move(scheduling_slice);
+    events_.emplace_back(std::move(event));
+  }
+
+  void OnCallstackSample(orbit_grpc_protos::CallstackSample callstack_sample) override {
+    orbit_grpc_protos::CaptureEvent event;
+    *event.mutable_callstack_sample() = std::move(callstack_sample);
+    events_.emplace_back(std::move(event));
+  }
+
+  void OnFunctionCall(orbit_grpc_protos::FunctionCall function_call) override {
+    orbit_grpc_protos::CaptureEvent event;
+    *event.mutable_function_call() = std::move(function_call);
+    events_.emplace_back(std::move(event));
+  }
+
+  void OnIntrospectionScope(orbit_grpc_protos::IntrospectionScope introspection_scope) override {
+    orbit_grpc_protos::CaptureEvent event;
+    *event.mutable_introspection_scope() = std::move(introspection_scope);
+    events_.emplace_back(std::move(event));
+  }
+
+  void OnGpuJob(orbit_grpc_protos::GpuJob gpu_job) override {
+    orbit_grpc_protos::CaptureEvent event;
+    *event.mutable_gpu_job() = std::move(gpu_job);
+    events_.emplace_back(std::move(event));
+  }
+
+  void OnThreadName(orbit_grpc_protos::ThreadName thread_name) override {
+    orbit_grpc_protos::CaptureEvent event;
+    *event.mutable_thread_name() = std::move(thread_name);
+    events_.emplace_back(std::move(event));
+  }
+
+  void OnThreadStateSlice(orbit_grpc_protos::ThreadStateSlice thread_state_slice) override {
+    orbit_grpc_protos::CaptureEvent event;
+    *event.mutable_thread_state_slice() = std::move(thread_state_slice);
+    events_.emplace_back(std::move(event));
+  }
+
+  void OnAddressInfo(orbit_grpc_protos::AddressInfo address_info) override {
+    orbit_grpc_protos::CaptureEvent event;
+    *event.mutable_address_info() = std::move(address_info);
+    events_.emplace_back(std::move(event));
+  }
+
+  void OnTracepointEvent(orbit_grpc_protos::TracepointEvent tracepoint_event) override {
+    orbit_grpc_protos::CaptureEvent event;
+    *event.mutable_tracepoint_event() = std::move(tracepoint_event);
+    events_.emplace_back(std::move(event));
+  }
+
+  void OnModulesUpdate(orbit_grpc_protos::ModuleUpdateEvent module_update_event) override {
+    orbit_grpc_protos::CaptureEvent event;
+    *event.mutable_module_update_event() = std::move(module_update_event);
+    events_.emplace_back(std::move(event));
+  }
+
+  [[nodiscard]] const std::vector<orbit_grpc_protos::CaptureEvent>& GetEvents() const {
```

Comment:
Again, according to the style guide, simple getters should be snake_case

Reply:
The styleguide says "may", i.e., it's a preference. Personally, I don't mind the snake_case when we have a struct or class that contains only or mostly data and there's almost a one-to-one relation between fields and methods (something like a proto), but I don't like it when that's not the case and the result is just a mix of styles.

Reply:
I am only concerned about the project-wide style. As far as I remember, by now, most simple getters are snake_case. But I am also not strong about this. I'll leave it to you. But we definitely already have classes where it's a big mixture of PascalCase and snake_case.

Reply:
> But we definitely already have classes where it's a big mixture of PascalCase and snake_case.

I know, I don't like it... In my opinion that should be reserved for "data" classes.

Take `GetChildPid` above for example. The reason why it would make sense to use `child_pid` is because you know that that's a field. But the field is private, so that's just an implementation detail, the class could have another way of retrieving that pid. While for data classes, it's normal that the accessor is based on a field.

In any case, there classes are private to the test file so it doesn't matter.